### PR TITLE
Move daemon image code in preparation for ImageService

### PR DESCRIPTION
**Pull request by dnephin**
Extract some code moving from #36240 so shrink that PR.

This PR just moves some functions into new files, and extracts a function for GetLayerFolders

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM :frog:

File: daemon/image_prune.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package daemon
+package daemon // import "github.com/docker/docker/daemon"
```
Comment (thaJeztah) [inline daemon/image_prune.go]: heh, should probably have been in the other commit 😅

File: daemon/image_windows.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/layer"
+	"github.com/docker/docker/pkg/system"
+	"github.com/pkg/errors"
+)
+
+// GetLayerFolders returns the layer folders from an image RootFS
+func (daemon *Daemon) GetLayerFolders(img *image.Image, rwLayer layer.RWLayer) ([]string, error) {
```
Comment (thaJeztah) [inline daemon/image_windows.go]: Does this have to be exported?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
one question, but looks good otherwise

File: daemon/image_windows.go (hunk 2 above)
Comment (mlaventure) [reply daemon/image_windows.go]: Good catch, it doesn't seem like it needs to be.

Comment (dnephin) [reply daemon/image_windows.go]: It will need to be in #36240 . So I can change it here and export it later, but that's probably not a good use of time.

**Review by mlaventure — APPROVED** [review]
LGTM

File: daemon/image_windows.go (hunk 2 above)
Comment (thaJeztah) [reply daemon/image_windows.go]: Ah, thanks for explaining 👍

**Review by thaJeztah — APPROVED** [review]
LGTM

File: daemon/image_windows.go (hunk 2 above)
Comment (dnephin) [reply daemon/image_windows.go]: Thanks for the review!

**Pull request merged**